packages/ui/src/composables/useInputMask/masks/parser.ts

```diff
@@ -0,0 +1,228 @@
+interface TokenBase {
```

Comment:
This parser is actually good. The only problem here is in `tree: Token[]` type. I should've been use TokenBase[] here, because `Token` is declared after `tree: Token[]`.

This also might be optimized if we parse normalized at the beginning. But it is too complex to understand.